python/python/lance/dataset.py

```diff
@@ -681,10 +682,21 @@ def version(self) -> int:
     @property
     def latest_version(self) -> int:
         """
-        Returns the lastest version of the dataset
+        Returns the latest version of the dataset.
         """
         return self._ds.latest_version()
 
+    def checkout_version(self, version) -> "LanceDataset":
+        """
+        Load the given version of the dataset.
+
+        This is a no-op if the dataset is already at the given version.
+        """
+        ds = copy.copy(self)
+        if version != ds.version:
+            ds._ds = self._ds.checkout_version(version)
```

Comment:
Uber nit: why not just lance.dataset(ds.uri, version=version)

Reply:
My understanding is that invalidates all the cached metadata (stored in the `Session` struct). My goal here is if another writer appends some new data, you can just call

```python
my_dataset = my_dataset.checkout_version(my_dataset.latest_version())
```

And then you will get an up-to-date reference, while keeping all the cached metadata you had before.